### PR TITLE
Fix Dockerfile to support Debian

### PR DESCRIPTION
**Pull request by zmichaels11**
### Issues
* #84 
* #80 (possibly)
* #81 

### Testing
* ran `test/run_e2e_test.sh`
* ran `test/run_e2e_test_debian.sh`

### CODE REVIEW

**Comment by zmichaels11** [comment]
This causes some errors if `ubuntu:latest` is used since `libgcc 7` and `libgcc 8` are both available

**Comment by emersonknapp** [comment]
This is good - i added a commit to properly source the ros dependencies that rosdep installed, if they exist

File: cross_compile/Dockerfile_ros2 [hunk 1]
```diff
@@ -100,14 +100,5 @@ RUN rosdep update && \
 RUN colcon mixin add cc_mixin \
     https://raw.githubusercontent.com/ros-tooling/cross_compile/master/mixins/index.yaml && \
     colcon mixin update cc_mixin
+RUN touch /opt/ros/${ROS_DISTRO}/setup.sh
```
Comment (piraka9011) [inline cross_compile/Dockerfile_ros2]: What is `setup.sh` is empty? How does this affect the next step?

**Comment by piraka9011** [comment]
Why is there a whole new `run_e2e_test_debian.sh` when we can just pass the `--os` flag?

File: cross_compile/Dockerfile_ros2 (hunk 1 above)
Comment (emersonknapp) [reply cross_compile/Dockerfile_ros2]: if it doesn't exist it gets created by this command as an empty file. if it does exist then its timestamp is updated.

if it was created as empty, the initial sourcing portion of the line below is a no-op

if it already existed, then it does its thing

Comment (emersonknapp) [reply cross_compile/Dockerfile_ros2]: however, the directory may not have existed already, in the "no-rosdeps" use case, so i'm adding a check for that

**Comment by zmichaels11** [comment]
> Why is there a whole new `run_e2e_test_debian.sh` when we can just pass the `--os` flag?
> Also, if we want that script to be explicitly for debian, should we remove the option to specify the OS in the argparser?

I didn't know that, removed e2e test and verified that `./run_e2e_test.sh --os debian` passes

**Pull request merged**